pkg/types/ccip/models.go

```diff
@@ -2,10 +2,49 @@ package ccip
 
 import (
 	"encoding/hex"
+	"fmt"
+	"strings"
+	"unicode"
+
+	"golang.org/x/crypto/sha3"
 )
 
 type Address string
 
+func (a *Address) UnmarshalJSON(bytes []byte) error {
```

Comment:
add a code comment about why we need this custom un/marshaling. iiuc, it boils down to backward compatibility